src/shared/port-provider.js

```diff
@@ -246,6 +221,35 @@ export class PortProvider {
     );
   }
 
+  /**
+   * Check that data and origin matches the expected values.
+   *
+   * @param {object} options
+   *   @param {Message} options.allowedMessage - the `data` must match this
+   *     `Message`.
+   *   @param {string} options.allowedOrigin - the `origin` must match this
+   *     value. If `allowedOrigin` is '*', the origin is ignored.
+   *   @param {any} options.data - the data to be compared with `allowedMessage`.
+   *   @param {string} options.origin - the origin to be compared with
+   *     `allowedOrigin`.
+   */
+  _messageMatches({ allowedMessage, allowedOrigin, data, origin }) {
+    if (allowedOrigin !== '*' && origin !== allowedOrigin) {
+      return false;
+    }
+
+    return isMessageEqual(data, allowedMessage);
+  }
+
+  /**
+   * @param {'frameConnected'} eventName
+   * @param {(source: 'guest'|'sidebar', port: MessagePort) => void} handler - this handler
+   *   fires when a request for the host frame has been granted.
```

Comment:
```suggestion
   *   fires when a frame connects to the host frame
```

The concept of requests and offers are really internal to PortProvider and PortFinder. Consumers only really need to know that a non-host frame connected to the host.

Reply:
👍🏼